src/hocs/withCheckNetworkConnection/withCheckNetworkConnection.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Alert } from 'react-native';
+import { useSelector } from 'react-redux';
+
+import { selectors } from 'app/state/electrumX';
+
+type checkNetworkConnectionArgs = (...args: any[]) => void;
+
+const withCheckNetworkConnection = <P extends Record<string, any>>(Component: React.ComponentType<P>) => (props: P) => {
+  const isInternetReachable = useSelector(selectors.isInternetReachable);
+  const isServerConnected = useSelector(selectors.isServerConnected);
+
+  const checkNetworkConnection = (func: checkNetworkConnectionArgs) => {
```

Comment:
wouldn't it be better to name it `callback` instead of `func`?

Reply:
changed to callback